### PR TITLE
Changed the accucor and isocorr validation file fields into a single field

### PR DESCRIPTION
**Pull request by hepcat72**
## Summary Change Description

- Improved the form's clean method so that errors would be associated with their fields.
- Added a check to the form's clean method to ensure that the files are excel and that the peak annotation files were specifically either accucor or isocorr (individually).
- Changed the order of the set_files args and made the sample arg optional.
- Added methods: is_accucor and is_isocorr to the AccucorDataLoader class.
- Changed the way things were imported from django.forms so that it would look cleaner.

## Affected Issues/Pull Requests

- Partially addresses #829
- Merges into PR #924
- Next PR: #925

## Review Notes

See comments in-line.

## Checklist

This pull request will be merged once the following requirements are met.  The
author and/or reviewers should uncheck any unmet requirements:

- Review requirements
  - Minimum approvals: 1 
  - No changes requested
  - All blocking issues resolved by reviewers
  - Specific reviewers: @__add_username_here__
    
  - Review period: 2 days 
- Associated issue/pull request requirements:
  
  - [x] All requirements in affected issues marked "resolved" are satisfied
  - [x] All required pull requests are merged *(or none)*
- Basic requirements
  
  - [x] [All linters pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#linting)
  - [x] [All tests pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#quality-control)
  - [x] All conflicts resolved
- Overhead requirements
  
  - [x] [New/changed method tests implemented/updated *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#test-implementation)
  - [ ] [Updated *Unreleased* section of `changelog.md` *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/changelog.md)
  - [x] [Migrations created & committed *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#migration-process)

### CODE REVIEW

File: DataRepo/forms.py [hunk 1]
```diff
@@ -1,13 +1,27 @@
...
+    ClearableFileInput,
+    FileField,
+    Form,
+    HiddenInput,
+    JSONField,
+    Select,
+    TextInput,
+    ValidationError,
+    formset_factory,
+)
```
Comment (hepcat72) [inline DataRepo/forms.py:16]: C. I just think it looks cleaner to have each one individually imported.

File: DataRepo/forms.py [hunk 2]
```diff
@@ -1,13 +1,27 @@
...
 
 from DataRepo.formats.dataformat import Format
 from DataRepo.formats.fluxcirc_dataformat import FluxCircFormat
 from DataRepo.formats.peakdata_dataformat import PeakDataFormat
 from DataRepo.formats.peakgroups_dataformat import PeakGroupsFormat
 from DataRepo.formats.search_group import SearchGroup
+from DataRepo.loaders.accucor_data_loader import AccuCorDataLoader
+from DataRepo.utils.file_utils import is_excel
```
Comment (hepcat72) [inline DataRepo/forms.py:25]: C. Skip down to my next comment.  All the rest is related to the import method change.

File: DataRepo/forms.py [hunk 3]
```diff
@@ -321,18 +335,16 @@ def __init__(self, *args, **kwargs):
...
 
-    animal_sample_table = forms.FileField(
-        required=False, widget=forms.ClearableFileInput(attrs={"multiple": False})
+    animal_sample_table = FileField(
+        required=False, widget=ClearableFileInput(attrs={"multiple": False})
     )
-    accucor_files = MultipleFileField(
-        required=False, widget=MultipleFileInput(attrs={"multiple": True})
-    )
-    isocorr_files = MultipleFileField(
```
Comment (hepcat72) [inline DataRepo/forms.py:330]: C. This is the meat

File: DataRepo/forms.py [hunk 4]
```diff
@@ -346,16 +358,52 @@ def clean(self):
         super().clean()
 
         study_doc = self.cleaned_data.get("animal_sample_table", None)
```
Comment (hepcat72) [inline DataRepo/forms.py:360]: C. I decided to validate that the files are excel and that the peak annotation files were either isocorr or accucor (based on sheet names, which I added as a method in the AccucorDataLoader class (and imported above).

Comment (hepcat72) [reply DataRepo/forms.py:360]: I also learned how to associate validation errors with the fields displayed in the form.

File: DataRepo/views/upload/validation.py [hunk 5]
```diff
@@ -52,50 +55,67 @@ class DataValidationView(FormView):
...
+            raise ProgrammingError(
+                f"The number of peak annotation file names [{len(peak_annotation_filenames)}] must be equal to the "
+                f"number of peak annotation files [{peak_annotation_files}]."
+            )
+
+        not_peak_annot_files = []
+        self.accucor_files = []
+        self.isocorr_files = []
+        self.accucor_filenames = []
+        self.isocorr_filenames = []
```
Comment (lparsons) [inline DataRepo/views/upload/validation.py:99]: minor: Just noticing this type of pattern of parallel lists. Perhaps using a dict where the keys are the filenames and the values are the files would keep things more "tidy"? Just a thought.

**Review by lparsons — APPROVED** [review]
Looks good. I especially like associating the form errors with the form fields. One small comment, feel free to ignore.

File: DataRepo/views/upload/validation.py (hunk 5 above)
Comment (hepcat72) [reply DataRepo/views/upload/validation.py:99]: Yeah, that's a valid observation.  Here, I was simply sticking with the previous pattern and trying to make minimal updates, as there is at least one method that takes these as arguments, though I will try and resist falling into that pattern in new code.

**Pull request merged**